core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -563,7 +556,7 @@ private static MetadataUpdate readRemoveSnapshots(JsonNode node) {
     MetadataUpdate metadataUpdate;
     if (snapshotIds.size() == 1) {
       Long snapshotId = Iterables.getOnlyElement(snapshotIds);
-      metadataUpdate = new MetadataUpdate.RemoveSnapshot(snapshotId);
+      metadataUpdate = new MetadataUpdate.RemoveSnapshots(snapshotId);
```

Comment:
I think we can initialize `RemoveSnapshots` without the `snapshotIds` size check now. 
```java
  private static MetadataUpdate readRemoveSnapshots(JsonNode node) {
    Set<Long> snapshotIds = JsonUtil.getLongSetOrNull(SNAPSHOT_IDS, node);
    Preconditions.checkArgument(
        snapshotIds != null,
        "Invalid set of snapshot ids to remove: must be non-null",
        snapshotIds);
    return new MetadataUpdate.RemoveSnapshots(snapshotIds);
  }
```